File: balancetree/balancetree.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func toBinaryTree(items []interface{}) *TreeNode {
	n := len(items)
	if n == 0 {
		return nil
	}
	return inner(0, items)
}

func inner(index int, items []interface{}) *TreeNode {
	n := len(items)
	if n <= index || items[index] == nil {
		return nil
	}
	return &TreeNode{
		Val:   items[index].(int),
		Left:  inner(2*index+1, items),
		Right: inner(2*index+2, items),
	}
}

func main() {
	n := toBinaryTree([]interface{}{1, 2, 3, 4, 5, 6, 7, 8, 9})
	println(n.Val)
}

File: balancetree/balancetree.py
class TreeNode:
    def __init__(self, val: int, left=None, right=None) -> None:
        self.val = val
        self.left = left
        self.right = right

    def __repr__(self) -> str:
        return f"val: {self.val}, left: {self.left}, right: {self.right}"

    def __str__(self) -> str:
        return str(self.val)

def to_binary_tree(items)->TreeNode:
    n = len(items)
    if n == 0 :
        return None
    def inner(index:int = 0) -> TreeNode:
        if n <= index or items[index] is None:
            return None
        node = TreeNode(items[index])
        node.left = inner(2 * index + 1)
        node.right = inner(2 * index + 2)
        return node
    return inner()

if __name__ == "__main__":
    root = to_binary_tree([1, 2, 3, None, None, 4, 5])
    print(root.__repr__())